### PR TITLE
detect-itype: add tests

### PR DESCRIPTION
**Pull request by hsadia538**
Previous PR #970 

Redmine [Ticket](https://redmine.openinfosecfoundation.org/issues/5590)

Describe Changes:

- Update Readme file
- Update `test.yaml `to add `min-version`

### CODE REVIEW

File: tests/detect-itype/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+PCAP
+====
+PCAP comes from the redmine ticket [5590](https://redmine.openinfosecfoundation.org/issues/5590)
+
+Redmine ticket
+==============
+https://redmine.openinfosecfoundation.org/issues/5590
+
+
```
Comment (inashivb) [inline tests/detect-itype/README.md:13]: nit: unnecessary newlines

File: tests/detect-itype/test.rules [hunk 2]
```diff
@@ -0,0 +1,5 @@
+alert icmp any any -> any any (itype:10; sid:1;)
+alert icmp any any -> any any ((itype:<15; sid:2;)
+alert icmp any any -> any any (itype:>20; sid:3;)
+alert icmp any any -> any any (itype:8<>20; sid:4;)
+
```
Comment (inashivb) [inline tests/detect-itype/test.rules:5]: nit: unnecessary newline

File: tests/detect-itype/test.rules [hunk 3]
```diff
@@ -0,0 +1,5 @@
+alert icmp any any -> any any (itype:10; sid:1;)
+alert icmp any any -> any any ((itype:<15; sid:2;)
```
Comment (inashivb) [inline tests/detect-itype/test.rules:2]: extra brace in rule language

File: tests/detect-itype/test.rules [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+alert icmp any any -> any any (itype:>20; sid:3;)
+alert icmp any any -> any any (itype:8<>20; sid:4;)
```
Comment (inashivb) [inline tests/detect-itype/test.rules:4]: Could you try and make the rules with the same itype conditions but such that all of them alert on the pcap?

File: tests/detect-itype/test.yaml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+requires:
+  min-version: 7
```
Comment (inashivb) [inline tests/detect-itype/test.yaml:2]: Is this needed here? Do these rules not work on Suricata version < 7?
You can fetch suricata-6.0.x and see if the tests pass on that too on your local system. 😉

**Review by inashivb — CHANGES_REQUESTED** [review]
Good job! :)
Some small changes requested inline.

File: tests/detect-itype/test.yaml (hunk 5 above)
Comment (hsadia538) [reply tests/detect-itype/test.yaml:2]: I added this based on your comment on the other suricata-verify PR so I thought maybe its  must for all s-v PRs now.

Comment (hsadia538) [reply tests/detect-itype/test.yaml:2]: Should I just remove it altogather?

File: tests/detect-itype/test.rules (hunk 4 above)
Comment (hsadia538) [reply tests/detect-itype/test.rules:4]: I added these rules based on what I found in the corresponding unit test. 
Should I alter the pcap to match the rule conditions or rules to match the pcap that I got. (The pcap I have right now has `itype 0 and 8`)
What's the better approach in this scenario?

File: tests/detect-itype/test.yaml (hunk 5 above)
Comment (jufajardini) [reply tests/detect-itype/test.yaml:2]: Have you tried running these tests on suricata-6.0.x, to see if they pass? If they do, then you can remove it, yes :)

File: tests/detect-itype/test.rules (hunk 4 above)
Comment (jufajardini) [reply tests/detect-itype/test.rules:4]: I would say that it's better to adjust the rules to the pcap you have, ensuring that all of the rules match on this PR.

File: tests/detect-itype/test.yaml (hunk 5 above)
Comment (hsadia538) [reply tests/detect-itype/test.yaml:2]: How can I build an older version to test?

Comment (jufajardini) [reply tests/detect-itype/test.yaml:2]: There's no need to build an older version for this, it's actually about checking out a branch with https://github.com/OISF/suricata/tree/master-6.0.x and running your suricata-verify from it :)

(But if you wanted to have a test for 6, only, for instance, you could have:)
```yaml
requires:
    version: 6
```

Comment (hsadia538) [reply tests/detect-itype/test.yaml:2]: got it, thanks. I will try and then update

Comment (hsadia538) [reply tests/detect-itype/test.yaml:2]: tests pass on version 6.0.x so I am removing the min version line

**Comment by hsadia538** [comment]
Followed by #979

**Pull request closed without merging**